### PR TITLE
Code quality

### PR DESCRIPTION
**Pull request by lippserd**
Blocked by #312

### CODE REVIEW

File: pkg/icingaredis/utils.go [hunk 1]
```diff
@@ -53,6 +54,7 @@ func CreateEntities(ctx context.Context, factoryFunc contracts.EntityFactoryFunc
 	return entities, com.WaitAsync(g)
 }
 
+// SetChecksums concurrently concurrently streams from the given entities and sets their checksums using the specified map and streams the results on a returned channel.
```
Comment (julianbrost) [inline pkg/icingaredis/utils.go:57]: Concurrency² for even more speed?

**Review by julianbrost — CHANGES_REQUESTED** [review]
The remaining comments in `pkg/icingadb/db.go` (line 378 and below) could probably all be a bit more verbose. Didn't want to comment each on its own.

File: pkg/icingaredis/utils.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+// CreateEntities streams and creates entities from the given Redis field value paris using the specified factory function, and streams them on a returned channel.
```
Comment (julianbrost) [inline pkg/icingaredis/utils.go:15]: Typo in pairs

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: pkg/icingadb/db.go [hunk 3]
```diff
@@ -373,9 +407,14 @@ func (db *DB) CreateStreamed(ctx context.Context, entities <-chan contracts.Enti
 	sem := db.getSemaphoreForTable(utils.TableName(first))
 	stmt, placeholders := db.BuildInsertStmt(first)
 
-	return db.NamedBulkExec(ctx, stmt, 1<<15/placeholders, sem, forward, nil)
+	return db.NamedBulkExec(ctx, stmt, utils.Min(1<<14/placeholders, 1<<10), sem, forward, nil)
```
[the remark this answers is not in the document]
Comment (julianbrost) [reply pkg/icingadb/db.go:410]: Created a review for that in the actual PR for this change: https://github.com/Icinga/icingadb/pull/312#pullrequestreview-705350715

File: pkg/driver/driver.go [hunk 4]
```diff
@@ -72,6 +70,26 @@ func (d Driver) OpenConnector(name string) (driver.Connector, error) {
 	}, nil
 }
 
+// Register makes our database Driver available under the name "icingadb-mysql".
+func Register(logger *zap.SugaredLogger) {
+	sql.Register("icingadb-mysql", &Driver{ctxDriver: &mysql.MySQLDriver{}, Logger: logger})
+	_ = mysql.SetLogger(mysqlLogger(func(v ...interface{}) { logger.Debug(v) }))
```
[the remark this answers is not in the document]
Comment (lippserd) [reply pkg/driver/driver.go:76]: True!

File: pkg/icingadb/db.go [hunk 5]
```diff
@@ -181,6 +200,12 @@ func (db *DB) BulkExec(ctx context.Context, query string, count int, sem *semaph
 	return g.Wait()
 }
 
+// NamedBulkExec bulk executes queries with named placeholders.
+// Takes in up to the number of entities specified in count from the arg stream,
+// and executes the query with this set of arguments until the arg stream has been processed.
+// The queries are executed in a separate goroutine with a weighting of 1
+// and can be executed concurrently to the extent allowed by the semaphore passed in sem.
+// Entities for which the query ran successfully will be streamed on the succeeded channel.
```
[the remark this answers is not in the document]
Comment (lippserd) [reply pkg/icingadb/db.go:208]: Your assumption is correct, although these are implementation details and should not be relevant in the docs. So what exactly is difficult to understand? Should I add something like "Derives and executes a new query with placeholders  automatically expanded and replaced with the appropriate DB driver's bindvar type."?

File: pkg/icingadb/db.go [hunk 6]
```diff
@@ -253,6 +273,12 @@ func (db *DB) NamedBulkExec(
 	return g.Wait()
 }
 
+// NamedBulkExecTx bulk executes queries with named placeholders in separate transactions.
+// Takes in up to the number of entities specified in count from the arg stream,
+// executes a new query for each entity in the set of arguments in a new transaction,
```
[the remark this answers is not in the document]
Comment (lippserd) [reply pkg/icingadb/db.go:278]: Yep.

File: pkg/icingadb/db.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (julianbrost) [reply pkg/icingadb/db.go:208]: I don't see a generic way to take any query and execute it on multiple sets of arguments except just running the query multiple times. But that's what `NamedBulkExecTx` does and this function performs no fallback to that one.

So there has to be some restriction on the types of queries this function can handle.

I do understand that this function just inherits its behavior from [`sqlx.DB.NamedExecContext()`](https://pkg.go.dev/github.com/jmoiron/sqlx#DB.NamedExecContext) but I also find that documentation lacking. At least to me, it's not clear that you're even supposed to pass a slice as `arg`.

**Comment by lippserd** [comment]
Removed some commits here and created #340 and #341 as these changes were not really related to code quality.

**Review by julianbrost — APPROVED** [review]

**Pull request merged**